paraatm/bdn/_bdn.py

```diff
@@ -0,0 +1,209 @@
+        model.compile(loss='mse',metrics=['mse'],optimizer=optimizer_)
+        return model
+    def _get_dnn_model(self):
+        """
+        Construct a deep neural network (dnn) model which is then passed to 'pred' method if flag_2 =='dnn'.
+        A dnn model is used to make predictions for input data which has linear or non-linear relationships with the output data, or target.
+        
+        
+        For more information about dnn model, please refer to:
+        https://en.wikipedia.org/wiki/Deep_learning#Deep_neural_networks
+        
+        Returns
+        -------
+        A sequential dnn model ready to be fit in 
+        """
+        if len(self.x_train.shape)!=2:
+            raise Exception('expect a 2 dimension array!')
+        in_shape = self.x_train.shape[-1]
+        model=Sequential()
+        model.add(Dense(self.num_unit,kernel_regularizer=l2(self.weight_decay),
+                       bias_regularizer=l2(self.weight_decay),
+                        kernel_initializer='random_uniform',
+                        input_dim=in_shape))
+        
+        model.add(Activation('relu'))
+        model.add(Dropout(self.idrop))
+        model.add(Dense(self.num_unit,kernel_regularizer=l2(self.weight_decay),
+                       bias_regularizer=l2(self.weight_decay),
+                        kernel_initializer='random_uniform',
+                       ))
+        model.add(Dropout(self.idrop))
+        model.add(Activation('relu'))  
+        if self.odrop:
+            model.add(Dropout(self.odrop))
+        model.add(Dense(1,activation='linear',
+                        kernel_regularizer=l2(self.weight_decay),
+                        bias_regularizer=l2(self.weight_decay)))
+        optimizer_=Adam(self.lr)
+        model.compile(loss='mse',metrics=['mse'],optimizer=optimizer_)
+        return model        
+        
+
+    def pred(self,flag_1 = True,flag_2 = 'rnn',custom_model = None):
+        """
+        pred is used to generate results for the deep learning model
+        
+        Parameters
+        ----------
+        flag_1: Boolen
+            By default: it's set as 'True'
+            flag_1 == True: prediction with uncertainty
+            flag_1 == False: deterministic result
+        
+        flag_2: str
+            By default, it's set as 'rnn'
+            flag_2 == 'rnn': generate results for the embedded rnn model
+            flag_2 == 'dnn': generate results for the embedded dnn model
+            flag_2 == 'custom': generate results for the customized model
+            When flag_2 == 'custom', a customized model must be provided
+        
+        custom_model: None is set by default, otherwise is a  customized sequential model
+        
+        Returns
+        -------
+        An array which is predictions for the test data
+        The array has a shape as num_test_data by num_iter
+        num_test_data:  Number of test data
+        num_iter: number of iterations
```

Comment:
Is the return type different when `flag_1` for is set for prediction with uncertainty?  If so, please document how the uncertainty is represented in the return type.

Reply:
Hi John,

A detailed explanation for how uncertainty is included is added to the document string.

Reply:
This doesn't actually address my original comment.  I cannot tell from reading the documentation how uncertainty is represented in the return type.  The question is not about the method used to calculate uncertainty, it is about the type of data that is returned from the `pred` function when the user has requested uncertainty via the flag (originally `flag_1`).

For example, is the return value a list that includes both a mean and a standard deviation?  Is it just a standard deviation?  Is it a variance?  Is it an array of means and standard deviations?  Is the return type different when different values are provided to `flag_1`?

Also, this description of the return type belongs in the `Returns` section of the docstring.  For an example, refer to the documentation for `SklearnGPRegression.__call__` in [gp.py](../blob/master/paraatm/rsm/gp.py).

Reply:
Hi John,

I just updated the module. A more detailed description of how uncertainty should be obtained is added to the document. 
pred_type has been renamed as get_var.
model.fit method has been moved to ___init__  so that the model only need to be fitted once for the deterministic and uncertainty case.
x_test and y_test parameter have been moved to 'pred' method, so that the user can reuse the model for different test data  at 'pred' method.